src/Native/Unix/System.Native/pal_io.c

```diff
@@ -176,6 +176,12 @@ static void ConvertFileStatus(const struct stat_* src, FileStatus* dst)
     dst->BirthTime = 0;
     dst->BirthTimeNsec = 0;
 #endif
+
+#if HAVE_STAT_FLAGS
+    dst->UserFlags = src->st_flags;
```

Comment:
I'm not sure if it applies to UF_HIDDEN, but in the past we've seen a lot of these defines have different values across platforms, which is one of the reasons we have these shims.  As such, if we stick with the UserFlags (see my other previous comment about that), we should define a PAL_ version in pal_io.h that has values which match the managed side, regardless of how the native platform defines it, and then here do the translation, e.g.
```C
dst->UserFlags = (src->st_flags & UF_HIDDEN == UF_HIDDEN) ? PAL_UF_HIDDEN : 0;
```

Reply:
Do we want to limit the flags only for the `UF_HIDDEN` one? As I went through the possible flags, I've noticed that you can also set `SF_ARCHIVED (0x00010000)` in OSX which can be mapped to `FileAttributes.Archived` and currently it is not. Is it ok to limit it currently only to `UF_HIDDEN` for now and if e.g `SF_ARCHIVED` was to be implemented in the future, then the one implementing it should change the implementation?

Reply:
> Is it ok to limit it currently only to UF_HIDDEN for now and if e.g SF_ARCHIVED was to be implemented in the future, then the one implementing it should change the implementation?

Yup.  Let's stick with just hidden for now.